lib/beaker/hypervisor/abs.rb

```diff
@@ -13,15 +13,39 @@ def initialize(hosts, options)
       @resource_hosts = JSON.parse(resource_hosts)
     end
 
+    def connection_preference(host)
+      vmhostname = host[:vmhostname]
+      if vmhostname && host[:hypervisor] == 'abs'
+        @resource_hosts.each do |resource_host|
+          if resource_host['hostname'] == vmhostname
+            engine = resource_host['engine']
+            case engine
+            when /^(vmpooler|nspooler)$/
+              # putting ip last as its not set by ABS
+              return [:vmhostname, :hostname, :ip]
+            else
+              super
+            end
+          end
+        end
+      end
+      super
```

Comment:
But maybe that can't happen; `connection_preference` won't ever be called in that case, since `Hypervisor.set_ssh_connection_preference(...)` won't exist so there's nothing that will call this method.

I was thinking we were going to have to guard against this method not existing, like we did below for the other method call.